### PR TITLE
UDM-ISS PR for Turtle Creek

### PR DESCRIPTION
**Pull request by gblewis1**

### CODE REVIEW

File: inbm/dispatcher-agent/dispatcher/dispatcher_class.py [hunk 1]
```diff
@@ -60,17 +61,8 @@
...
-    """Return the config path for this agent, taken by default from LOGGERCONFIG environment
-    variable and then from a fixed default path.
-    """
-    try:
-        return os.environ['LOGGERCONFIG']
-    except KeyError:
-        return DEFAULT_LOGGING_PATH
-
+# Mutex lock
+sql_lock = Lock()
```
Comment (gblewis1) [inline inbm/dispatcher-agent/dispatcher/dispatcher_class.py:65]: move sql code into another PR

File: inbm/dispatcher-agent/dispatcher/dispatcher_class.py [hunk 2]
```diff
@@ -94,15 +86,9 @@ def _check_type_validate_manifest(xml: str,
...
-        log_config_path = get_log_config_path()
-        msg = f"Looking for logging configuration file at {log_config_path}"
-        print(msg)
-        fileConfig(log_config_path,
-                   disable_existing_loggers=False)
-
         self._dispatcher_broker = broker
         self._install_check_service = install_check_service
-        self.update_queue: Queue[Tuple[str, str]] = Queue(1)
+        self.update_queue: Queue[Tuple[str, str, Optional[str]]] = Queue(1)
```
Comment (gblewis1) [inline inbm/dispatcher-agent/dispatcher/dispatcher_class.py:91]: we could turn this Tuple into a class for clarity

File: inbm/dispatcher-agent/dispatcher/dispatcher_class.py [hunk 3]
```diff
@@ -715,15 +706,47 @@ def check_dispatcher_state_info(self) -> None:
...
+            schedule = ScheduleManifestParser(manifest, schedule_manifest_schema, manifest_schema)
+        except XmlException as e:
+            logger.error("XMLException parsing schedule: " + str(e))
+            if request_id:
+                dispatcher._send_result(f"Error parsing schedule manifest: {str(e)}", request_id)
+            return
+        
+        # TODO: Change single and repeated to add the schedules to the scheduler DB
+        if schedule.single_scheduled_requests or schedule.repeated_scheduled_requests:
+            sql_lock.acquire()
```
Comment (gblewis1) [inline inbm/dispatcher-agent/dispatcher/dispatcher_class.py:734]: When using the sqlite3 module in Python, you typically don't need to manually set locks. SQLite automatically handles locking and concurrency control internally.

(also would be good to move this to another PR)

**Review by nmgaston — APPROVED** [review]

**Review by nmgaston — APPROVED** [review]

File: inbm/dispatcher-agent/dispatcher/dispatcher_class.py (hunk 1 above)
Comment (nmgaston) [reply inbm/dispatcher-agent/dispatcher/dispatcher_class.py:65]: Decided not to move to another PR.

File: inbm/dispatcher-agent/dispatcher/dispatcher_class.py (hunk 2 above)
Comment (nmgaston) [reply inbm/dispatcher-agent/dispatcher/dispatcher_class.py:91]: Noted.  Will do in next PR.
